client/components/events/EventViewToggle/EventViewToggle.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useEffect } from "react";
+import { useRouter } from 'next/router'
+import { EventCalendarView } from "../EventCalendarView"
+import { EventListView } from "../EventListView/EventListView"
+import { Event } from '../../../context'
+import { colours, device, fontWeight, desktopFontSize, mobileFontSize } from '../../../styles'
+import styled from 'styled-components'
+
+enum View {
+  list = 'list',
+  calendar = 'calendar',
+}
+
+interface EventViewProps {
+  filteredEvents: Array<Event>
+}
+
+const EventViewToggleContainer = styled.div<any>`
+  margin: 14px 24px;
+  @media not all and ${device.laptop} {
+    margin: 10%;
+  }
+`
+
+const EventViewToggleButtonsContainer = styled.div<any>`
+  display: flex;
+  width: 100%;
+  margin-bottom: 10px;
+  @media ${device.laptop} {
+    margin-top: 100px;
+  }
+`
+
+const EventViewToggleButtonsSpacer = styled.div<any>`
+  flex-shrink: 0;
+  flex-basis: 200px;
+  @media ${device.tablet} {
+    display: none;
+  }
+`
+
+const EventViewToggleButtons = styled.div<any>`
+  display: flex;
+  margin-bottom: 50px;
+  cursor: pointer;
+`
+
+const ToggleText = styled.div<any>`
+  ${(props: any) => props.isActiveView && `
+    font-weight: bold;
+  `};
+  margin-right: 25px;
+`
+
+export const EventViewToggle = ({ filteredEvents }: EventViewProps) => {
+  const router = useRouter()
+
+  const { view } = router.query;
+
+  useEffect(() => {
+    //refresh component whenever query param changes
+  }, [router.query.view])
+
+  const updateActiveView = (view) => {
+    router.push({query: {
+      view: view
+    }})
+  }
+
+  let formattedView = View.list
+  if(typeof view === 'string'){
+    formattedView = View[view]
+  }
+  
+  return (
+    <EventViewToggleContainer> 
+      <EventViewToggleButtonsContainer>
+        <EventViewToggleButtonsSpacer/>
+        <EventViewToggleButtons>
+          <EventViewToggleButton activeView={formattedView} viewType={View.list} updateActiveView={updateActiveView}/>
+          <EventViewToggleButton activeView={formattedView} viewType={View.calendar} updateActiveView={updateActiveView}/>
+        </EventViewToggleButtons>
+      </EventViewToggleButtonsContainer>
+      <ActiveView activeView={formattedView} filteredEvents={filteredEvents}/>
+    </EventViewToggleContainer>
+  );
+};
+
+interface ActiveViewProps {
+  activeView: View,
+  filteredEvents: Array<Event>
+}
+
+const ActiveView = ({activeView, filteredEvents}: ActiveViewProps) => {
+  
+  const getActiveView = () => {
+    switch(activeView){
```

Comment:
```suggestion
    switch (activeView) {
```

Wait, Prettier was okay with not putting spaces?